src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -1,81 +1,328 @@
 
 class SignUpModalController {
   /* @ngInject */
-  constructor (gettext, sessionService) {
-    this.gettext = gettext
+  constructor ($log, $scope, $rootScope, $location, $window, sessionService, cartService, orderService, envService) {
+    this.$log = $log
+    this.$scope = $scope
+    this.$rootScope = $rootScope
+    this.$location = $location
+    this.$window = $window
     this.sessionService = sessionService
+    this.orderService = orderService
+    this.cartService = cartService
+    this.imgDomain = envService.read('imgDomain')
+    this.publicCru = envService.read('publicCru')
   }
 
   $onInit () {
-    this.modalTitle = this.gettext('Sign Up')
-    this.isLoading = false
-    this.setPristine()
-  }
+    if (includes([Roles.identified, Roles.registered], this.sessionService.getRole())) {
+      this.onStateChange({ state: 'sign-in' })
+    }
 
-  signUp () {
-    if (!this.signUpForm.$valid) {
-      return
+    this.setUpSignUpWidget()
+
+    if (!this.isInsideAnotherModal) {
+      this.cartCount = 0
+      this.getTotalQuantitySubscription = this.cartService.getTotalQuantity().subscribe(count => {
+        this.cartCount = count
+      }, () => {
+        this.cartCount = 0
+      })
     }
-    // https://github.com/CruGlobal/thekey/wiki/Self-Service-REST-API#create-user
-    // https://github.com/CruGlobal/cortex_gateway/wiki/Create-User
-    this.setPristine()
+
     this.isLoading = true
-    this.sessionService
-      .signUp(this.email, this.password, this.first_name, this.last_name)
-      .subscribe((data) => {
-        this.isLoading = false
-        this.onSuccess(data)
-      }, (error) => {
-        this.hasError = true
-        switch (error.status) {
-          case 400:
-            if (error.data && error.data.error === 'invalid_password') {
-              this.signUpErrors.invalid_password = true
-              this.invalidPasswordMessage = error.data.description
-            } else {
-              this.signUpErrors[error.status] = true
+    this.submitting = false
+  }
+
+  $onDestroy () {
+    this.oktaSignInWidget.remove()
+  }
+
+  async setUpSignUpWidget () {
+    const donorData = await this.loadDonorDetails()
+    this.$window.currentStep = 1 // Default to step 1
+
+    this.oktaSignInWidget = new OktaSignIn({
+      ...this.sessionService.oktaSignInWidgetDefaultOptions,
+      assets: {
+        baseUrl: `${this.$window.location.origin}/assets/okta-sign-in/`
+      },
+      flow: 'signup',
+      registration: {
+        parseSchema: (schema, onSuccess) => {
+          const step = this.$window.currentStep || 1
+          // Split the form into multiple steps for better user experience.
+          // Retain the values entered by the user in each step.
+          // Pre-populate the form fields with existing user details to save time.
+          const steps = {
+            // Step 1: Name and email
+            1: [
+              {
+                ...schema[0],
```

Comment:
What do you think about referencing fields by their name instead of their index?

```suggestion
                ...schema.find(field => field.name === 'userProfile.firstName'),
```

Reply:
That makes sense. We can leave it as-is.